runtime/runtime-params-estimator/estimator-warehouse/src/db.rs

```diff
@@ -0,0 +1,196 @@
+//! Database connection and struct representing rows in the data tables.
+
+use std::path::Path;
+
+use chrono::NaiveDateTime;
+use rusqlite::{params, Connection, Row};
+
+use crate::Metric;
+
+/// Wrapper around database connection
+pub(crate) struct DB {
+    conn: Connection,
+}
+
+impl DB {
+    /// Opens an existing SQLite DB or creates it
+    pub(crate) fn open(path: &Path) -> anyhow::Result<Self> {
+        let conn = Connection::open(path)?;
+        let init_sql = include_str!("init.sql");
+        conn.execute_batch(init_sql)?;
+        Ok(Self { conn })
+    }
+    #[cfg(test)]
+    pub(crate) fn test() -> Self {
+        let conn = Connection::open_in_memory().unwrap();
+        let init_sql = include_str!("init.sql");
+        conn.execute(init_sql, []).unwrap();
+        Self { conn }
+    }
+    pub(crate) fn insert_estimation(&self, row: &EstimationRow) -> anyhow::Result<()> {
```

Comment:
It's a bit inconsistent that other methods are on `EstimationRow`, but this one is on DB. I guess the difference is read-vs-update? Probably doesn't matter much either way.